File: Source/Aladdin.CAPI.GOST/ASN1/GOST/GOSTR3410PrivateKeyValueInfo.cs
using System; 
using System.Runtime.Serialization; 

namespace Aladdin.ASN1.GOST
{
    [Serializable]
	public class GOSTR3410PrivateKeyValueInfo : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<OctetString>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<OctetString>().Factory(), Cast.N), 
		}; 
		// конструктор при сериализации
        protected GOSTR3410PrivateKeyValueInfo(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public GOSTR3410PrivateKeyValueInfo(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public GOSTR3410PrivateKeyValueInfo(OctetString privateKeyMaskValue, 
			OctetString publicKeyValue) : base(info, privateKeyMaskValue, publicKeyValue) {}

		public OctetString PrivateKeyMaskValue { get { return (OctetString)this[0]; } } 
		public OctetString PublicKeyValue      { get { return (OctetString)this[1]; } }
	}
}

File: Source/Aladdin.CAPI.GOST.Java/src/aladdin/asn1/gost/GOSTR3410PrivateKeyValueInfo.java
package aladdin.asn1.gost;
import aladdin.asn1.*; 
import java.io.*; 

///////////////////////////////////////////////////////////////////////////////
// GostR3410-2001-KeyValueMask ::= OCTET STRING;
// GostR3410-2001-PublicKey    ::= OCTET STRING {PubKeyX | PubKeyY};
// GostR3410-2001-KeyValueInfo ::= SEQUENCE{
// 	GostR3410-2001-KeyValueMask,
// 	GostR3410-2001-PublicKey 
// }
///////////////////////////////////////////////////////////////////////////////
public final class GOSTR3410PrivateKeyValueInfo extends Sequence<OctetString>
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(OctetString.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(OctetString.class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public GOSTR3410PrivateKeyValueInfo(IEncodable encodable) throws IOException { super(encodable, info); }

    // конструктор при закодировании
	public GOSTR3410PrivateKeyValueInfo(OctetString privateKeyMaskValue, 
		OctetString publicKeyValue)
    {
        super(info, privateKeyMaskValue, publicKeyValue); 
    }
    public final OctetString privateKeyMaskValue() { return get(0); } 
	public final OctetString publicKeyValue     () { return get(1); }
}
